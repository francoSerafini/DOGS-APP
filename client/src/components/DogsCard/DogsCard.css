.card {
    width: 80%;
    border: solid;
    background-color: rgba(114, 107, 107, 0.877);
    border-radius: 50px;
    margin-bottom: 20px;
}

.image {
    width: 80%;
    height: 60%;
    border-radius: 20px;
} 

.image:hover {
    transform: scale(1.5);
  }

  @media only screen and (max-width:800px) {
    
    .card {
        width: 100%;
    }
  }