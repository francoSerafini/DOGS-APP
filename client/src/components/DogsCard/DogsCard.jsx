import React, { Component } from 'react';
import './DogsCard.css';

export class DogsCard extends Component {

    render() {
        return (
            <div id={this.props.id} onClick={this.props.onClick}>
                <h2 id={this.props.id}>{this.props.name}</h2>
                <img className='image' src={this.props.image} alt='Not Found' id={this.props.id}></img>
                {this.props.temperaments &&
                <div id={this.props.id}>
                    <h3 id={this.props.id} >Temperaments: {this.props.temperaments}</h3>
                    <h3 id={this.props.id}>Weight: {this.props.weight} Kg</h3>
                </div>
                }
            </div>
        );
    };
};