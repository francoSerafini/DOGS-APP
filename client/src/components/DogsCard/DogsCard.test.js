import '@testing-library/jest-dom';
import { configure, shallow } from 'enzyme';
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import  DogsCard  from './DogsCard';
import isReact from 'is-react';

configure({ adapter: new Adapter() });

describe('<DogDetail />', () => {
    let dogsCard;
    const dog = {
        id:1,
        name:"Affenpinscher",
        weight:"3 - 6",
        temperaments:"Stubborn, Curious, Playful, Adventurous, Active, Fun-loving",
        image:"https://cdn2.thedogapi.com/images/BJa4kxc4X.jpg"
    };

beforeEach(() => {
    dogsCard = (dog) =>
        shallow(
            <DogsCard
                key={dog.id} 
                name={dog.name} 
                image={dog.image}
                temperaments={dog.temperaments} 
                weight={dog.weight}
            />
        );
    expect(isReact.classComponent(DogsCard)).toBeFalsy();
});

it('Should render <DogsCard/> correctly', () => {
    expect(dogsCard).toMatchSnapshot();
});

it('Should render an "h2" containing breed of dog', () => {
    expect(dogsCard(dog).find('h2').text()).toBe(
        dog.name
    );
});

it('Should render an "img" tag and use the image of the dogs breed as the source', () => {
    expect(dogsCard(dog).find('img').prop('src')).toEqual(
        dog.image
    );
})

it('Should render an "h3" containing the text "Temperaments: " plus the dogs breed temperaments', () => {
    expect(dogsCard(dog).find('h3').at(0).text()).toBe(
        `Temperaments: ${dog.temperaments}`
    );
});

it('Should render an "h3" containig the text "Weight: " plus the dogs breed weight', () => {
    expect(dogsCard(dog).find('h3').at(1).text()).toBe(
        `Weight: ${dog.weight}Kg`
    )
})

});