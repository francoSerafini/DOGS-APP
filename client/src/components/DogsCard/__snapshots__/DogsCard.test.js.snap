// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogDetail /> Should render <DogsCard/> correctly 1`] = `[Function]`;
